app-server/app/src/main/scala/token/Routes.scala

```diff
@@ -0,0 +1,102 @@
+package com.azavea.rf.token
+
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.headers.{Authorization, GenericHttpCredentials}
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.HttpMethods._
+import akka.http.scaladsl.model.StatusCodes._
+import akka.http.scaladsl.model.Uri.{Path, Query}
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.unmarshalling.Unmarshal
+
+import spray.json.DefaultJsonProtocol
+
+import com.azavea.rf.auth.Authentication
+import com.azavea.rf.utils.{Config, UserErrorHandler}
+
+
+/**
+  * Routes for tokens
+  */
+trait TokenRoutes extends Authentication
+    with UserErrorHandler
+    with Config
+    with SprayJsonSupport
+    with DefaultJsonProtocol {
+
+  import com.azavea.rf.AkkaSystem._
+
+  case class DeviceCredential(id: String, device_name: String)
+  implicit val deviceCredentialJson = jsonFormat2(DeviceCredential)
+
+  case class AuthorizedToken(id_token: String, expires_in: Int, token_type: String)
+  implicit val authorizedTokenJson = jsonFormat3(AuthorizedToken)
+
+  case class RefreshToken(refresh_token: String)
+  implicit val refreshTokenJson = jsonFormat1(RefreshToken)
+
+  val uri = Uri(s"https://$auth0Domain/api/v2/device-credentials")
+  val headers = List(
+    Authorization(GenericHttpCredentials("Bearer", auth0Bearer))
+  )
+
+  def tokenRoutes: Route = pathPrefix("api" / "tokens") {
+    handleExceptions(userExceptionHandler) {
```

Comment:
I think it makes sense to split up the token routes and stack them such that we do something like [`thumbnailRoutes`](https://github.com/azavea/raster-foundry/blob/develop/app-server/app/src/main/scala/thumbnail/Routes.scala#L28) to make this a little more clearer.